src/write_BCs/write_boundary_conditions.py

```diff
@@ -233,4 +208,38 @@ def apply_tropomi_operator_to_one_tropomi_file(filename):
                 ds.to_netcdf(os.path.join(config["workDir"], "blended-boundary-conditions", os.path.basename(filename)))
             else:
                 print(f"Writing to {os.path.join(config['workDir'], 'tropomi-boundary-conditions', os.path.basename(filename))}")
-                ds.to_netcdf(os.path.join(config["workDir"], "tropomi-boundary-conditions", os.path.basename(filename)))
\ No newline at end of file
+                ds.to_netcdf(os.path.join(config["workDir"], "tropomi-boundary-conditions", os.path.basename(filename)))
+
+
+if __name__ == "__main__":
+
+    # Arguments from run_boundary_conditions.sh
+    blendedTROPOMI = (sys.argv[1] == "True") # use blended data?
+    satelliteDir = sys.argv[2] # where is the satellite data?
+    # Start of GC output (+1 day except 1 Apr 2018 because we ran 1 day extra at the start to account for data not being written at t=0)
+    start_time_of_interest = np.datetime64(datetime.datetime.strptime(sys.argv[3], "%Y%m%d"))
+    if start_time_of_interest != np.datetime64("2018-04-01T00:00:00"):
+        start_time_of_interest += np.timedelta64(1, "D")
+    # End of GC output
+    end_time_of_interest = np.datetime64(datetime.datetime.strptime(sys.argv[4], "%Y%m%d"))
+    print(f"\nwrite_boundary_conditions.py output for blendedTROPOMI={blendedTROPOMI}")
+    print(f"Using files at {satelliteDir}")
+
+    """
+    This script works in three parts, utilizing the GEOS-Chem output from run_boundary_conditions.sh
+    (1) Make a gridded (2.0 x 2.5 x daily) field of TROPOMI/GEOS-Chem co-locations
+        - for every TROPOMI observation, apply the TROPOMI operator to the GEOS-Chem fields
+        - average the TROPOMI XCH4 and GEOS-Chem XCH4 to a (2.0 x 2.5) grid for each day
+    (2) Make a gridded (2.0 x 2.5 x daily) field of the bias between TROPOMI and GEOS-Chem
+        - subtract the TROPOMI and GEOS-Chem grids from part 1 to get a starting point for the bias
+        - smooth this field spatially (5 lon grid boxes, 5 lat grid boxes) then temporally (15 days backwards)
+        - fill NaN values with the latitudinal average at that time
+            - for a latitudinal average to be defined, there must be >= 30 grid cells at that latitude
+            - when a latitudinal average cannot be found, the closest latitudinal average is used
+    (3) Write the boundary conditions
+        - using the bias from Part 2, subtract the (GC-TROPOMI) bias from the GC boundary conditions
+    """
+
+    daily_means = part1(satelliteDir, start_time_of_interest, end_time_of_interest)
+    bias = part2(daily_means)
+    part3(bias)
```

Comment:
Can you make the function names descriptive? Something like: `create_daily_means()`, `calculate_bias()`, and `write_bias_corrected_files()`